zerver/lib/cache.py

```diff
@@ -69,18 +76,21 @@ def get_or_create_key_prefix():
 
     return prefix
 
-KEY_PREFIX = get_or_create_key_prefix()
+KEY_PREFIX = get_or_create_key_prefix() # type: str
 
 def bounce_key_prefix_for_testing(test_name):
+    # type: (str) -> None
     global KEY_PREFIX
     KEY_PREFIX = test_name + ':' + str(os.getpid()) + ':'
 
 def get_cache_backend(cache_name):
+    # type: (str) -> get_cache
     if cache_name is None:
         return djcache
     return get_cache(cache_name)
 
 def cache_with_key(keyfunc, cache_name=None, timeout=None, with_statsd_key=None):
+    # type: ignore # TODO: https://www.python.org/dev/peps/pep-0484/#callable
```

Comment:
for cache_with_key function's first argument no. of argument is inconsistent.
I tried with Callback[[Any], Any] but getting
`zerver/models.py:607: error: Argument 1 to "cache_with_key" has incompatible type Callable[[str, Any], str]; expected Callable[[Any], Any]`
With Callable[[Any, Any], Any] I am getting
`zerver/models.py:265: error: Argument 1 to "cache_with_key" has incompatible type Callable[[Any], Any]; expected Callable[[Any, Any], Any]`
I tried with variable no. of arguments as Callable[..., Any] but It is giving 
```
zerver/models.py:60: error: Cannot infer type of lambda
zerver/models.py:1153: error: Cannot infer type of lambda
```
@timabbott  can you help me here, Thanks